src/GovUk.Education.ExploreEducationStatistics.Content.Model/ContentBlock.cs

```diff
@@ -98,7 +100,27 @@ public DataBlock()
 
         public ObservationQueryContext Query { get; set; }
 
-        public List<IChart> Charts { get; set; }
+        [JsonIgnore]
+        public List<IChart> Charts
+        {
+            get
+            {
+                return ChartsInternal.Select(chart =>
+                {
+                    if (chart.Title.IsNullOrEmpty())
+                    {
+                        chart.Title = Heading;
+                    }
+                    return chart;
+                }).ToList();
+
+            }
+            set => ChartsInternal = value;
+        }
+
+        [JsonProperty("Charts")]
```

Comment:
So this gets included in JSON even though it's private?  I would have thought that the one you actually wanted output in JSON would be "Charts", as that's the one that's had the Titles filled in for the frontend's benefit?

Reply:
I'll make the switch - so I think this means that in the database, if a chart inherits a table heading, the chart title won't be null anymore - I guess it doesn't really matter either way, but since the frontend expects the chart title to be explicitly provided regardless of whether it is inherited, I think saving it this way in the database also makes sense.

Reply:
I'm having a change of heart with this. So after making this change, now when the frontend sends a null chart title (if the chart is inheriting the table heading), in C# memory, the chart title is null, but when this gets serialized into JSON, it goes through the getter and therefore the chart title isn't null but is explicitly set as the table heading.

This difference between what the chart title is in the C# programs memory vs. how it is stored in the DB might be a little confusing - so maybe its best if they're the same to make it simpler to understand.

Thoughts?

Reply:
I think the initial confusion for me was that I assumed that the JSON annotations were for the benefit of converting to JSON for the front end.  But it turns out it's for the benefit of storing it as JSON in the db instead, in a form that is consistent with the in-memory C# representation.  Am I right?

In which case, I think all we need is a note by this JSONProperty annotation to explain that it's for the purposes of storing this object "pure" in the database.  Any idea where we do this out of interest?

Reply:
Discussed this with Duncan on Slack. I've added a comment that hopefully clarifies why it is done the way its done. Now, the saved chart JSON in the database will be the same as the chart title as it is saved in program memory.

For anyone who happens to read this in the future, the chart JSON is currently stored in the ContentBlocks table, under the DataBlock_Charts column - `Charts` is mapped to this in ContentDbContext.